### ISSUE TITLE
Cleanup public interface of the library

### ISSUE THREAD

**Issue by goodmanmd** [HeadspringLabs/bulk-writer#45]
The following classes are all marked `public` but should be `internal` instead.  Move them to the Internal folder and change the accessibility level as appropriate:

- `GetPropertyValueHandler`
- `MappingDestination`
- `MappingSource`
- `PropertyMapping`
